xbmc/dbwrappers/dataset.h

```diff
@@ -406,6 +402,49 @@ class Dataset  {
   const sql_record* const get_sql_record();
 
  private:
+ 
+  unsigned int indexMapID;
+
+  struct IndexMapEntry    //Struct to store a indexMapped field access entry
```

Comment:
sorry to nit-pick here, but we should add "field" to the variable names. IndexMap is a bit to generic IMO. `FieldIndexMapEntry` `FieldIndexMapComparator` etc.